src/main/java/org/jitsi/videobridge/EndpointMessageTransport.java

```diff
@@ -0,0 +1,517 @@
+/*
+ * Copyright @ 2017 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge;
+
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+import org.json.simple.*;
+import org.json.simple.parser.*;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * @author Boris Grozev
+ */
+class EndpointMessageTransport
```

Comment:
Yes. Endpoint will provide a "sendMessage" method, and EndpointMessageTransport will decide which of the underlying transport to send it over. For incoming messages, it will handle the JSON parsing (and e.g. call Endpoint#selectedEndpointsChanged(List<Endpoint>))